Data/Aeson/Types/Internal.hs

```diff
@@ -349,6 +353,20 @@ liftP2 f pa pb = Parser $ \path kf ks ->
     (\a -> runParser pb path kf (\b -> ks (f a b)))
 {-# INLINE liftP2 #-}
 
+accumulateSequenceList :: [Parser a] -> Parser [a]
+accumulateSequenceList = accumulateTraverseList id
+
+accumulateTraverseList :: (a -> Parser b) -> [a] -> Parser [b]
+accumulateTraverseList f s = case s of
+  []    -> pure mempty
+  h : t -> (:) <$> (f h) <*>+ (accumulateTraverseList f t)
+
+accumulateTraverseVector :: (a -> Parser b) -> Vector a -> Parser (Vector b)
+accumulateTraverseVector f v =
+    if V.null v
```

Comment:
I wonder, if it makes sense to define

```haskell
newtype AccParser a = AccParser { runAccParser :: Parser a }

instance Applicative AccParser where
    AccParser f <*> AccParser x = AccParser (f <*>+ x) -- !!!
```

and use it, and not re-implement `traverse` for everything.

Reply:
I agree, it's a good idea to use `Traversable` here. Reexporting `Parser` versions that use `AccParser` under the hood is nice too.